client/my-sites/email/paths.ts

```diff
@@ -0,0 +1,189 @@
+		if ( ! domainName.startsWith( ':' ) ) {
+			// Encodes domain names so addresses with slashes in the path (e.g. used in site redirects) don't break routing.
+			// Note they are encoded twice since page.js decodes the path by default.
+			domainName = encodeURIComponent( encodeURIComponent( domainName ) );
+		}
+
+		return (
+			resolveRootPath( relativeTo ) +
+			'/' +
+			domainName +
+			'/' +
+			slug +
+			'/' +
+			siteName +
+			buildQueryString( urlParameters )
+		);
+	}
+
+	if ( siteName ) {
+		return '/email/' + siteName + buildQueryString( urlParameters );
+	}
+
+	return '/email';
+}
+
+// Retrieves the URL of the Add New Mailboxes page for email forwarding
+export const getAddEmailForwardsPath: EmailPathUtilityFunction = (
+	siteName,
+	domainName,
+	relativeTo,
+	urlParameters
+) => getPath( siteName, domainName, 'forwarding/add', relativeTo, urlParameters );
+
+// Retrieves the URL of the Add New Mailboxes page either for G Suite or Google Workspace
+export function getAddGSuiteUsersPath(
+	siteName: string | null | undefined,
+	domainName: string | null | undefined,
+	productType: string,
+	relativeTo?: string,
+	source?: string
+) {
+	if ( siteName && domainName ) {
+		return getPath( siteName, domainName, productType + '/add-users', relativeTo, {
+			source,
+		} );
+	}
+
+	if ( siteName ) {
+		return '/email/' + productType + '/add-users/' + siteName;
+	}
+
+	return '/email';
+}
+
+export const getManageTitanAccountPath: EmailPathUtilityFunction = (
+	siteName,
+	domainName,
+	relativeTo,
+	urlParameters
+) => getPath( siteName, domainName, 'titan/manage', relativeTo, urlParameters );
+
+export const getManageTitanMailboxesPath: EmailPathUtilityFunction = (
+	siteName,
+	domainName,
+	relativeTo,
+	urlParameters
+) => getPath( siteName, domainName, 'titan/manage-mailboxes', relativeTo, urlParameters );
+
+export const getNewTitanAccountPath: EmailPathUtilityFunction = (
+	siteName,
+	domainName,
+	relativeTo,
+	urlParameters
+) => getPath( siteName, domainName, 'titan/new', relativeTo, urlParameters );
+
+// Retrieves the URL to set up Titan mailboxes
+export const getTitanSetUpMailboxPath: EmailPathUtilityFunction = (
+	siteName,
+	domainName,
+	relativeTo,
+	urlParameters
+) => getPath( siteName, domainName, 'titan/set-up-mailbox', relativeTo, urlParameters );
+
+export const getTitanSetUpThankYouPath = (
+	siteName: string | null | undefined,
+	domainName: string | null | undefined,
+	emailAddress?: string | null,
+	relativeTo?: string
+) =>
+	getPath(
+		siteName,
+		domainName,
+		'titan/set-up-mailbox/thank-you',
+		relativeTo,
+		emailAddress ? { email: emailAddress } : {}
+	);
+
+export const getTitanControlPanelRedirectPath: EmailPathUtilityFunction = (
+	siteName,
+	domainName,
+	relativeTo,
+	urlParameters
+) => getPath( siteName, domainName, 'titan/control-panel', relativeTo, urlParameters );
+
+export const getEmailManagementPath: EmailPathUtilityFunction = (
```

Comment:
A better name, consistent with the new convention that removes the `emailManagement` prefixes from the path functions, would be `getManagePath`.

Reply:
I consciously tried to mention `email`, `titan`, `gsuite` or `forwarding` in all function names to make them easier for other devs to understand at-a-glance. In this case, I think the current name adds clarity and doesn't break the new convention in any outrageous way (whereas `getManagementPath` is very generic and would require devs to look up which file the function is imported from).